### ISSUE TITLE
Ionic Framework 1.2 support

### ISSUE THREAD

**Issue by JLNNN** [aliok/ion-affix#12]
I know you're officially not maintaining this project any longer, but do you think you can update it to work with the latest Ionic Framework update v1.2?

Thanks in advance,
  JLNNN

**Comment by ace-han** [comment]
+1

**Comment by J-who** [comment]
I'm still getting scrollTop is undefined with that fix. Looks like event.originalEvent doesn't contain a .detail

**Comment by J-who** [comment]
Has anyone had any luck fixing this issue?

**Comment by seasonstar** [comment]
This issue has been fixed, thanks @genu.

**Comment by seasonstar** [comment]
It works on ionic 1.2.4

**Comment by J-who** [comment]
Yeah it's working for me on ios & web with 1.2.4, but not android with and without crosswalks.

**Comment by lookitsatravis** [comment]
FYI if anyone comes looking here to solve this issue for Android, it can be solved by disabling Native Scrolling on the ion-scroll that you want to ion-affix with. Native Scroll is awesome, but it breaks anything that was depending on "scrollTop" from the Ionic scroll event. To disable globally:

``` js
$ionicConfigProvider.scrolling.jsScrolling(true);
```

Otherwise, you can do:

``` html
<ion-scroll overflow-scroll="false">
  <!-- list or whatever -->
</ion-scroll>
```

More: http://blog.ionic.io/native-scrolling-in-ionic-a-tale-in-rhyme/

### Reason

The error is at line 182 where ion-affix tries to read the "scrollTop" property of the event. The native scrolling even does not include that information (specifically, the "originalEvent" and "detail" properties are missing on a native scroll event). So, simply enabling JS scroll on that item fixes it. I've got a crap android device, and the scroll still seems pretty smooth using XWalk.

Hope this helps someone!

**Comment by andrecbr** [comment]
Changing

``` javascript
var scrollTop = (event.detail || event.originalEvent && event.originalEvent.detail).scrollTop;
```

to

``` javascript
var scrollTop = event.target.scrollTop;
```

~~It works (scrolltop position), but the native scrolling not working yet, because fixed elements is not "showing", but "behind the scene" (on top) the process is ok.~~ (really? I don't know!)
I tryed changing this line `angular.element($ionicScroll.element).append(clone);` using after(clone), but without success.

PS: sorry my english :)

**Comment by darron1217** [comment]
I usually fix those kind of scroll problem by using code below.
(This supports both native and js scroll)

`var event = event.originalEvent || event;`
`var scrollTop = event.detail ? event.detail.scrollTop : event.target.scrollTop;`

**Comment by yacaeh** [comment]
Wow thank you so much @darron1217 It works on other scroll problems too!
